shared/chat/conversation/list/container.js

```diff
@@ -3,81 +3,83 @@ import * as Constants from '../../../constants/chat'
 import * as Creators from '../../../actions/chat/creators'
 import HiddenString from '../../../util/hidden-string'
 import ListComponent from '.'
-import {List, Map} from 'immutable'
+import {List} from 'immutable'
+import {compose} from 'recompose'
 import {connect} from 'react-redux'
+import {createSelector} from 'reselect'
 import {downloadFilePath} from '../../../util/file'
 import {navigateAppend} from '../../../actions/route-tree'
-import {pick} from 'lodash'
-import {compose} from 'recompose'
-import {getPath} from '../../../route-tree'
-import {chatTab} from '../../../constants/tabs'
 
 import type {OpenInFileUI} from '../../../constants/kbfs'
-import type {Options} from '../messages'
-import type {Props, OptionsFn} from '.'
 import type {OwnProps, StateProps, DispatchProps} from './container'
+import type {Props} from '.'
 import type {TypedState} from '../../../constants/reducer'
 
+// TODO change this. This is a temporary store for the messages so we can have a function to map from
+// messageKey to Message to support the 'edit last message' functionality. We should change how this works
+let _messages: List<Constants.Message> = List()
+const _getMessageFromMessageKey = (messageKey: Constants.MessageKey): ?Constants.Message => _messages.find(m => m.key === messageKey)
+
+const getPropsFromConversationState = createSelector(
+  [Constants.getSelectedConversationStates, Constants.getSelectedInbox, Constants.getSupersedes],
+  (conversationState, inbox, _supersedes) => {
+    let supersedes = null
+    let messageKeys = List()
+    let validated = false
+    if (conversationState) {
+      if (!conversationState.moreToLoad) {
+        supersedes = _supersedes
+      }
+
+      messageKeys = conversationState.messages.map(m => m.key)
+      _messages = conversationState.messages
+      validated = inbox && inbox.state === 'unboxed'
+    }
+    return {
+      messageKeys,
+      supersedes,
+      validated,
+    }
+  }
+)
+
+// This is a temporary solution until I can cleanup the reducer in a different PR
+// messageKeys is being derived so it can change even when nothing else is causing re-renders
+// As a short term 'cheat' i'm keeping the last copy and returning that if its equivalent. TODO take this out later
+let _lastMessasgeKeys = List()
```

Comment:
but there were lots of times where messageKeys was effectively the same list but being re-derived so causing a wasted render